src/ows/src/main/java/org/geoserver/ows/util/RequestUtils.java

```diff
@@ -63,7 +63,7 @@ public static String getVersionPreOws(List<String> providedList, List<String> ac
         if (acceptedList == null || acceptedList.isEmpty()) return provided.last().toString();
 
         // next figure out what the client accepts (and check they are good version numbers)
-        TreeSet<Version> accepted = new TreeSet<>();
+        List<Version> accepted = new ArrayList<>();
```

Comment:
This change, along with the method javadoc,  prompted me to look at the specifications better.

As you can see, RequestUtils has two negotiation methods, one for pre-OWS 1.1 services, and one for OWS 1.1+ services, while ``getVersionOws11`` should be used for those based on OWS 1.1.

AcceptVersions was introduced in OWS 1.1, some services simply do not have such parameter in their spec: WMS 1.1 and 1.3, WFS 1.0, WCS 1.0 just don't have it, and have their own negotiation mechanism that is different from what OWS 1.1 mandates. In particular, in those protocols "version" is the parameter to be used, and the algorithm is the following (picked from WMS 1.3):

> 
>  *  If a version unknown to the server and higher than the lowest supported version is requested, the server shall send the highest version it supports that is less than the requested version.
> * If a version lower than any of those known to the server is requested, then the server shall send the lowest version it supports.
> * If the client does not support the version sent by the server, it may either cease communicating with the server or send a new request with a different version number that the client does support. 

WFS 1.1 is a strange beast, in that it has an ``acceptVersions`` parameter, but still mandates the above version negotiation algorithm.

Since OWS 1.1 instead, it works differently, again from the spec:

> The server, upon receiving a GetCapabilities request, shall scan through this list and find
the first version number that it supports. It shall then return a service metadata document
conforming to that version of the specification, and containing that value of the ―version‖
parameter. If the list does not contain any version numbers that the server supports, the
server shall return an Exception with exceptionCode="VersionNegotiationFailed".

Long story short, we need to find a mechanism to figure out which of the two negotiation mechanism to use. Checking if AcceptVersions is present is tempting, but the WFS 1.1 seems to break that approach... wondering if we can simply ignore it and treat it as if it was a OWS 1.1 service....